### PR TITLE
Promote edk2, future, mtools; add python-pefile, python-virt-firmware

### PR DESCRIPTION
**Pull request by vinceaperri**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Moves edk2 to core for the kubevirt golden container.

###### Change Log  

- Move edk2 to core and upgrade to 20230301gitf80f052277c8
- Move mtools to core for edk2 and upgrade to 4.0.43
- Add python-virt-firmware 23.5 for edk2
- Add python-pefile 2023.2.7 for python-virt-firmware
- Move future to core for python-pefile and upgrade to 0.18.3

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: 370812

### CODE REVIEW

File: SPECS/edk2/edk2.spec [hunk 1]
```diff
@@ -0,0 +1,1086 @@
...
+BuildRequires:  xorriso
+
+# For generating the variable store template with the default certificates
+# enrolled.
+BuildRequires:  python3-virt-firmware >= 23.5
+
+# endif build_ovmf
+%endif
+
+%if %{cross}
```
Comment (0xba1a) [inline SPECS/edk2/edk2.spec:139]: Are we supporting cross-compilers in Mariner?
If not can we remove this section completely?

File: SPECS/edk2/edk2.spec [hunk 2]
```diff
@@ -0,0 +1,1086 @@
...
+%{_bindir}/PatchPcdValue
+%{_bindir}/TargetTool
+%{_bindir}/Trim
+%{_bindir}/UPT
+%dir %{_datadir}/%{name}
+%{_datadir}/%{name}/Python
+
+
+%changelog
+* Fri May 26 2023 Vince Perri <viperri@microsoft.com> - 20230301gitf80f052277c8-32
```
Comment (0xba1a) [inline SPECS/edk2/edk2.spec:675]: Please update cgmanifest.json with correct version

File: SPECS/future/future.spec [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+%{_bindir}/futurize-%{python3_version}
+%{_bindir}/pasteurize-%{python3_version}
+%{python3_sitelib}/future/
+%{python3_sitelib}/past/
+%{python3_sitelib}/libfuturize/
+%{python3_sitelib}/libpasteurize/
+%{python3_sitelib}/*.egg-info
+
+%changelog
+* Tue May 30 2023 Vince Perri <viperri@microsoft.com> - 0.18.2-6
```
Comment (0xba1a) [inline SPECS/future/future.spec:112]: Please update correct version in cgmanifest.json

File: SPECS/edk2/edk2.spec (hunk 1 above)
Comment (vinceaperri) [reply SPECS/edk2/edk2.spec:139]: No. Sure I can just remove it.

File: SPECS/edk2/edk2.spec (hunk 2 above)
Comment (vinceaperri) [reply SPECS/edk2/edk2.spec:666]: Done

**Review by 0xba1a — APPROVED** [review]
LGTM

**Pull request merged**